anoma/namada-interface
# feat/550 - Fix chain ID issues in Extension

resolves #550 

### Testing

Setup & Notes:
- Set up local node, and accounts in extension with balances in local node
- If you don't have balances in testnet, that's fine because tx don't work there either, but you should see the balance eventually update to zero after switching to public testnet
- _Optional_ You can remove the `NAMADA_INTERFACE_NAMADA_CHAIN_ID` and `...NAMADA_URL`, but it doesn't matter, the form will work and just override them - if you remove it, it will default to current public testnet

Testing:

1. With the interface connected to extension, in extension, click the _Settings_ icon, then `Network`
2. Change between the following `Chain ID` and `URL`:
  `public-testnet-15.0dacadb8d663`
  `https://proxy.heliax.click/public-testnet-15.0dacadb8d663/`

  _OR_

  `${CHAIN_ID}` (whatever your local chain ID is)
  `http://localhost:27657` or `http://127.0.0.1:27657` - you can ignore the proxy
3. You should see that the interface re-fetches balances for that chain. When using your local node, transfers should work (_NOTE_ Testnet transfers are not working right now) - if you have balances on testnet, you will see those (eventually, it is slow)
  
### Screenshots

_Update network form_
<img width="371" alt="Screen Shot 2024-01-04 at 10 23 02 AM" src="https://github.com/anoma/namada-interface/assets/330911/75a247f8-f3ea-4747-b15f-a6ba3871b8c8">
